.github/workflows/build.yml

```diff
@@ -0,0 +1,214 @@
+name: build
+
+# Controls when the action will run
+on:
+
+  # Trigger the workflow on all pushes, except on tag creation
+  push:
+    branches:
+    - '**'
```

Comment:
Can we restrict the runs on push to the `main` branch only, please? Pull requests are already enabled below, as well as manual workflow dispatches for early testing.

Reply:
I disagree with that. Having the tests run by themselves for any commit when experimenting on a branch is very useful and should not be deactivated. It's better to be aware of a problem as soon as possible than at the final step when opening a pull request.
And having to run the tests by hand seems to defeat the purpose of continuous integration.

Reply:
Yes I would like to keep it for the public GH-hosted runners please.
And I understand you don't want to overload your GPU-queue.